tritium-metrics/src/main/java/com/palantir/tritium/metrics/CacheMetrics.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.RatioGauge;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+public final class CacheMetrics {
+
+    private CacheMetrics() {}
+
+    public static void register(TaggedMetricRegistry taggedMetrics, Stats stats, String cacheName) {
+        InternalCacheMetrics.createMetrics(stats, InternalCacheMetrics.taggedMetricName(cacheName))
+                .forEach(taggedMetrics::registerWithReplacement);
+    }
+
+    static void forEachMetric(CacheMetrics.Stats stats, BiConsumer<String, Gauge<?>> consumer) {
+        consumer.accept("cache.estimated.size", stats.estimatedSize());
```

Comment:
Would be nice to make all of these `Optional` so we don't have to report metrics we know are garbage.